### ISSUE TITLE
Rpresto 1.4.5 "drop table command" gave an error message!! (but successfully drop the table)

### ISSUE THREAD

**Issue by charlvit** [prestodb/RPresto#254]
Dear support,

> RPresto::dbFetch(RPresto::dbSendQuery(conn, "DROP TABLE if exists <schema>.<table_name>"), -1)
**Error in .combine_results(res_list) : 
  Chunk column names are different across chunks: []**

note: 
1) old version 1.4.1 seem to be ok (no error msg)
2) dbRemoveTable(conn, in_schema("<schema>","<table_name>")) gave same error message!!

Charlvit

**Comment by charlvit** [comment]
![rpresto145_error](https://github.com/prestodb/RPresto/assets/50406769/09fea52b-1190-45b5-9eb6-e909694fd947)

**Comment by jarodmeng** [comment]
Can you create a minimal reprex so I can help troubleshoot? We have a unit test on `dbRemoveTable()` (see https://github.com/prestodb/RPresto/blob/master/tests/testthat/test-dbRemoveTable.R) and it has been passing on my local machine. Having a reprex would greatly help identify what went wrong.

**Comment by charlvit** [comment]
[cilia-stag_reprex.md](https://github.com/prestodb/RPresto/files/11449695/cilia-stag_reprex.md)

Hi jarodmeng,
I have attached above cilia-stag_reprex.md file for your debug. Please change con to your local presto. Appreciate your help.

**Comment by charlvit** [comment]
note: the actual table was created and drop without problem, just only error msg.

**Comment by jarodmeng** [comment]
Thanks for the example and note. I tested the same code on my local machine using postgres and there is no error. My best guess so far is that the problem is contained with Hive. I don't have a local hive environment set up to test it at the moment. I will need more time looking into it.

**Comment by charlvit** [comment]
Thanks jarodmeng.

**Comment by copernican** [comment]
I have also seen this issue with RPresto 1.4.5 and DBI 1.1.3, against Trino (needs `use.one.query = TRUE`) with a Hive backend.

``` r
library(DBI)
library(RPresto)

conn <- dbConnect(
  drv = Presto(),
  use.trino.headers = TRUE,
  catalog = "my_catalog",
  schema = "tmp",
  user = "my_username",
  host = "my_host",
  port = 1234
)

tbl <- "mtcars_tmp"

dbExistsTable(conn, tbl)
#> [1] FALSE
dbWriteTable(conn, tbl, mtcars, use.one.query = TRUE)
dbExistsTable(conn, tbl)
#> [1] TRUE
dbRemoveTable(conn, tbl)
#> Error in .combine_results(res_list): Chunk column names are different across chunks: []
dbExistsTable(conn, tbl)
#> [1] FALSE
```

<sup>Created on 2023-06-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sorensje** [comment]
I have also been experiencing a similar error.  e.g.:

```
rs_fetch_drop <- RPresto::dbFetch(rs_send_drop, n = -1)
  query FINISHED [-------------------------------------------------]   0% in  2sError in .combine_results(res_list) : 
  Chunk column names are different across chunks: []
```

The problem seems to be related to result objects being returned from the Trino server. On my end, the server running trio version 406 gave this error. When I reverted to trino v352 , the error was resolved.

**Comment by jarodmeng** [comment]
> I have also been experiencing a similar error. e.g.:
> 
> ```
> rs_send_drop <- dbSendQuery(presto_conn, "DROP TABLE IF EXISTS u_sorensje.iris_test") #sending alone doesn't execute the drop
> rs_fetch_drop <- RPresto::dbFetch(rs_send_drop, n = -1)
>   query FINISHED [-------------------------------------------------]   0% in  2sError in .combine_results(res_list) : 
>   Chunk column names are different across chunks: []
> ```
> 
> The problem seems to be related to result objects being returned from the Trino server. On my end, the server running trio version 406 gave this error. When I reverted to trino v352 , the error was resolved.

Can you provide a reprex? I will try to reproduce the error in Trino and troubleshoot.

**Comment by ChristianHorenkamp** [comment]
I'm also struggling with this problem.

within the dbFetch method the variable res_list contains of three tibbles
[[1]]
# A tibble: 0 x 0

[[2]]
# A tibble: 0 x 0

[[3]]
# A tibble: 0 x 0

After calling purrr::keep(res_list, ~ (ncol(.) > 0)) in .combine_results the res_list ist empty which encounters the problem.

Therefore calling res_list <- purrr::keep(res_list, ~ (ncol(.) > 0)) before the check to preserve for empty data frames should fix the problem.

**Comment by jarodmeng** [comment]
This is weird. When the drop table command runs successfully, the `res_list` variable should contain a list of 3 tibbles, 2 of them are empty and the last one shouldn't be empty.

This is what my results are when running locally on my machine

```
Browse[3]> res_list
[[1]]
# A tibble: 0 × 0

[[2]]
# A tibble: 0 × 0

[[3]]
# A tibble: 1 × 1
  result
  <lgl> 
1 TRUE  
```

It looks like that it could be a database issue whereby it doesn't return the result output.

**Comment by willshen99** [comment]
> Thanks for the example and note. I tested the same code on my local machine using postgres and there is no error. My best guess so far is that the problem is contained with Hive. I don't have a local hive environment set up to test it at the moment. I will need more time looking into it.

I have a Trino have both hive and postgres catalog. `dbRemoveTable` doesn't work for either of these -- same error message.

I'm using Trino release 461. 


> It looks like that it could be a database issue whereby it doesn't return the result output.


I think this is probably the reason. As I looked into the query json output in Trino UI:
```
    "outputDataSize" : "0B",
    "failedOutputDataSize" : "0B",
    "outputPositions" : 0,
    "failedOutputPositions" : 0,
    "outputBlockedTime" : "0.00ns",
```


dbCreateTable (when creating an empty table) is also failing #278 with same error message. Checked Trino Ui, exactly same, nothing got outputted.

**Comment by jarodmeng** [comment]
Fixed by #283
